frontend/packages/operator-lifecycle-manager/integration-tests-cypress/views/operator.ts

```diff
@@ -0,0 +1,144 @@
+import { projectDropdown } from '@console/cypress-integration-tests/views/common';
+import { detailsPage } from '@console/cypress-integration-tests/views/details-page';
+import { listPage } from '@console/cypress-integration-tests/views/list-page';
+import { modal } from '@console/cypress-integration-tests/views/modal';
+import { nav } from '@console/cypress-integration-tests/views/nav';
+
+export const operator = {
+  install: (
```

Comment:
The install action should be decoupled from the assertion. What if we want to write a test case for an operator that should fail to install?

Reply:
Please see the use of `cy.intercept()` in [#9142](https://github.com/openshift/console/pull/9142/files#diff-888fcd3f03658ad6355c60586b25467f9f1ca3f95c21fc6869bb2b4f3531a2a3R104) to mock failed uninstallation.

I think it would be rather difficult to decouple assertions from commands in this case since we kinda want to test assertions as we go through the many steps of installing an operator or creating an operand instance.   At the moment we have two tests (OLM install to single and global namespaces), it we took out the assertions, they would have to be duplicated in each test suite which uses the operator view.

Reply:
Right, you can mock a failure, but you would still need to either create a new helper or just write the test case manually for that. You wouldn't be able to use your `operator.install` helper at all in order to test a failed install. IMO, it would be better if our helpers were less coupled to specific test cases so that they could be reused. If you wrote an install/uninstall helper that is agnostic to success or failure, then it could be reused in the story that you linked.